### PR TITLE
fix: Return http conflict when expression is invalid

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java [hunk 1]
```diff
@@ -34,5 +34,6 @@
 {
     OK,
     WARNING,
-    ERROR
+    ERROR,
+    CONFLICT
```
[the remark this answers is not in the document]
Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java:38]: is there a specific reason for not adding the 4th value? If this is something that we all have agreed upon for consistency reasons then I can revert it.

[the remark this answers is not in the document]
Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java:38]: Secondly sending response as HTTP.OK even when we have Status.ERROR does not make sense to me. Either we map this Status.ERROR to some meaning full HTTP response or we create this 4th value in Status. As a result of this, we will have Status.CONFLICT which can be mapped to HTTP.CONFLICT.

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java:38]: As discussed, `Status.ERROR` is used in combination with HTTP response status of `Conflict` as setup by `org.hisp.dhis.dxf2.webmessage.WebMessageUtils#conflict(java.lang.String)`.

[the remark this answers is not in the document]
Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/feedback/Status.java]: resolved

**Review by jbee — APPROVED** [review]

**Pull request merged**